src/sciline/pipeline.py

```diff
@@ -434,6 +433,26 @@ def __setitem__(self, key: Type[T], param: T) -> None:
             )
         self._set_provider(key, Provider.parameter(param))
 
+    def get_provider(
+        self, tp: Union[Type[T], Item[T]]
+    ) -> Union[Callable[[Any, ...], T], T]:
+        '''Get the provider that produces the type. If the type is provided by a
+        parameter the method returns the value of the parameter.'''
+        provider = self._get_unique_provider(tp, HandleAsBuildTimeException())[0]
+        if provider.kind == 'function':
+            return provider.func
```

Comment:
Because I agreed with the concern that the `Provider` class isn't currently exposed, and returning a `Provider` here would expose it. I think just returning the same type that was provided by the user keeps things simpler.

Reply:
I'm not concerned about exposing it. This can be useful in its own right. 

Your solution makes it difficult to distinguish between providers and parameters. You can't check the type of the return value of `get_provider` because parameters can be callable and providers can now be any callable object, not just functions. So you would have to compare the returned type to `tp`.

Reply:
My only concern about expose `Provider` is that it would increase the api surface area. I agree that it's a problem that you can't generally determine if the returned provider is a parameter or a function.